python/ray/ml/preprocessors/encoder.py

```diff
@@ -101,29 +104,61 @@ def __init__(self, label_column: str):
         self.label_column = label_column
 
     def _fit(self, dataset: Dataset) -> Preprocessor:
-        stats = dataset.aggregate(UniqueValues(self.label_column))
-        self.stats = {row: _sorted_value_indices(stats[row]) for row in stats}
+        self.stats_ = _get_unique_value_indices(dataset, self.label_column)
         return self
 
     def _transform_pandas(self, df: pd.DataFrame):
+        _validate_df(df, self.label_column)
+
         def column_label_encoder(s: pd.Series):
-            s_values = self.stats[f"unique_values({s.name})"]
+            s_values = self.stats_[f"unique_values({s.name})"]
             return s.map(s_values)
 
         df[self.label_column] = df[self.label_column].transform(column_label_encoder)
         return df
 
     def __repr__(self):
-        return f"<Encoder label column={self.label_column} stats={self.stats}>"
+        return f"<Encoder label column={self.label_column} stats={self.stats_}>"
+
+
+def _get_unique_value_indices(
+    dataset: Dataset, *columns: str
+) -> Dict[str, Dict[str, int]]:
+    results = {}
+    for column in columns:
+        values = _get_unique_values(dataset, column)
+        if None in values:
```

Comment:
not sure if this is robust enough, perhaps try `any(pd.isnull(x) for x in values)`